src/api/DeferredJobsApiClient.ts

```diff
@@ -1,93 +1,94 @@
 import { ProgressPromise } from '@prezly/progress-promise';
 
-import type { ApiResponse, HttpClient, Params, ParamsWithPayload } from '../http';
+import type { ApiResponse, Params, ParamsWithPayload } from '../http';
 import { HttpCodes, isDeferredJobResponse } from '../http';
 import { routing } from '../routing';
 import type { JobState } from '../types';
 import { JobStatus } from '../types';
 
 import type { ApiClient } from './ApiClient';
+
 const JOB_STATUS_POLLING_INTERVAL = 2000; // ms
 
 async function sleep(milliseconds: number): Promise<void> {
     return new Promise((resolve) => setTimeout(resolve, milliseconds));
 }
 
-async function handleDeferredJob<V = any, P = any>(
-    http: HttpClient,
+type Job<V, P> = {
+    id: string;
+    state: JobState<V, P>;
+};
+
+function handleDeferredJob<V = any, P = any>(
+    api: ApiClient,
     request: Promise<ApiResponse<V>>,
 ): ProgressPromise<V, P> {
-    const response = await request;
+    return new ProgressPromise<V, P>(async (resolve, reject, update) => {
+        const response = await request;
 
-    if (response.status === HttpCodes.ACCEPTED && isDeferredJobResponse(response.payload)) {
-        return new ProgressPromise<V, P>(async function (resolve, reject, progress) {
+        if (response.status === HttpCodes.ACCEPTED && isDeferredJobResponse(response.payload)) {
+            const id = response.payload.progress.id;
             do {
-                const response = await http.get<{ job: JobState<V, P> }>(routing.jobsUrl, {
```

Comment:
- Problem 1: `http.get()` did not prepend `baseUrl`
- Problem 2: To request job progress, we need to access `routing.jobsUrl/${id}`
- Problem 3: The response shape is different: `{ job: { id: string, state: JobState } }`